airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -134,24 +131,25 @@ public void accept(final String messageString, final Integer sizeInBytes) throws
    * @return PartialAirbyteMessage if the message is valid, empty otherwise
    */
   @VisibleForTesting
-  public static Optional<PartialAirbyteMessage> deserializeAirbyteMessage(final String messageString) {
+  public static PartialAirbyteMessage deserializeAirbyteMessage(final String messageString) {
     // TODO: (ryankfu) plumb in the serialized AirbyteStateMessage to match AirbyteRecordMessage code
     // parity. https://github.com/airbytehq/airbyte/issues/27530 for additional context
-    final Optional<PartialAirbyteMessage> messageOptional = Jsons.tryDeserialize(messageString, PartialAirbyteMessage.class)
-        .map(partial -> {
-          if (Type.RECORD.equals(partial.getType()) && partial.getRecord().getData() != null) {
-            return partial.withSerialized(partial.getRecord().getData().toString());
-          } else if (Type.STATE.equals(partial.getType())) {
-            return partial.withSerialized(messageString);
-          } else {
-            return null;
-          }
-        });
-
-    if (messageOptional.isPresent()) {
-      return messageOptional;
+    var partial = Jsons.tryDeserialize(messageString, PartialAirbyteMessage.class)
+        .orElseThrow(() -> new RuntimeException("Unable to deserialize PartialAirbyteMessage."));
+
+    final var msgType = partial.getType();
+    if (Type.RECORD.equals(msgType) && partial.getRecord().getData() != null) {
+      // store serialized json
+      partial.withSerialized(partial.getRecord().getData().toString());
+      // drop unmarshalled json
```

Comment:
nit: what about adding a comment here explaining we want this to stay as String to relieve memory pressure, and it's less complex to leverage Jackson to unmarshall the json and later drop it, rather than write a custom deserialiser?

Reply:
Done, I'm not sure that I fully get the jackson part.